### PR TITLE
CDAP-14314 fix upgrade tool

### PR DESCRIPTION
**Pull request by CuriousVini**
Build: https://builds.cask.co/browse/CDAP-RUT1626-1

image

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java [hunk 1]
```diff
@@ -370,6 +370,8 @@ public void persistSubscriberState(String topic, String messageId) {
   public static void setupDatasets(DatasetFramework datasetFramework) throws IOException,
     DatasetManagementException {
     datasetFramework.addInstance(JobQueueDataset.class.getName(), Schedulers.JOB_QUEUE_DATASET_ID,
-                                 DatasetProperties.EMPTY);
+                                 DatasetProperties.builder().
+                                   add(Constants.Scheduler.JOB_QUEUE_NUM_PARTITIONS, DEFAULT_NUM_PARTITIONS)
```
Comment (anew) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java:374]: why is this needed if it is the default?

Comment (anew) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java:374]: Oh got it. I made this mandatory to make sure this property is always set.

**Review by anew — APPROVED** [review]
LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java (hunk 1 above)
Comment (CuriousVini) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDataset.java:374]: Its because https://github.com/caskdata/cdap/blob/release/5.1/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/queue/JobQueueDatasetDefinition.java#L48-L50 - expects it to be present

**Comment by CuriousVini** [comment]
Already tested on cluster from release 5.0 -> release 5.1. Build is green. Merging.
